src/vaadin-dialog.html

```diff
@@ -289,6 +391,172 @@
+          };
+          overlay.addEventListener('mousedown', startDrag);
+        }
+
+        _addResizableEventListeners() {
+          const overlay = this.$.overlay.$.overlay;
+          const originalBounds = {top: 0, left: 0, width: 0, height: 0};
+          const originalMouse = {top: 0, left: 0};
+          const resize = (e, edge) => {
+            this._snapToPixels();
+            const minimumSize = 40;
+            edge.split('').forEach(direction => {
+              switch (direction) {
+                case 'n': {
+                  const height = originalBounds.height - (e.pageY - originalMouse.top);
+                  const top = originalBounds.top + (e.pageY - originalMouse.top);
+                  if (height > minimumSize) {
+                    this._setBounds({top, height});
+                  }
+                  break;
+                }
+                case 'e': {
+                  const width = originalBounds.width + (e.pageX - originalMouse.left);
+                  if (width > minimumSize) {
+                    this._setBounds({width});
+                  }
+                  break;
+                }
+                case 's': {
+                  const height = originalBounds.height + (e.pageY - originalMouse.top);
+                  if (height > minimumSize) {
+                    this._setBounds({height});
+                  }
+                  break;
+                }
+                case 'w': {
+                  const width = originalBounds.width - (e.pageX - originalMouse.left);
+                  const left = originalBounds.left + (e.pageX - originalMouse.left);
+                  if (width > minimumSize) {
+                    this._setBounds({left, width});
+                  }
+                  break;
+                }
+              }
+            });
+          };
+          const stopResize = edge => window.removeEventListener('mousemove', resizeEvents[edge]);
+          const resizeEvents = {};
+          const stopResizeEvents = {};
+
+          // Note: edge controls added before corners
+          ['n', 'e', 's', 'w', 'nw', 'ne', 'se', 'sw'].forEach(edge => {
+            const resizer = document.createElement('div');
+            if (edge.length === 1) {
+              resizer.classList.add('edge');
+            }
+            resizer.classList.add('resizer');
+            resizer.classList.add(edge);
+            resizer.addEventListener('mousedown', e => {
+              e.preventDefault();
+              e.stopPropagation();
+              this._moveOverlayToTop();
+              const overlayBounds = overlay.getBoundingClientRect();
+              const containerBounds = this.$.overlay.getBoundingClientRect();
+              originalBounds.top = overlayBounds.top - containerBounds.top;
+              originalBounds.left = overlayBounds.left - containerBounds.left;
+              originalBounds.width = overlayBounds.width;
+              originalBounds.height = overlayBounds.height;
```

Comment:
This code is repeated in 3 places. Could create a shared method for it like `_getOverlayBounds() {...}`

Reply:
Done.